ITL-6 Add sorting functionality

BiroAron opened the pull request.

BiroAron committed:
ITL 24 - Implement make  a component and six button first

kincsokiss commented on src/components/header/ArrowButton.vue:
```diff
@@ -0,0 +1,10 @@
+</template>
+
+<script setup lang="ts">
+//import UpArrowIcon from '../icons/UpArrowIcon.vue'
```
Please remove the unused import.

kincsokiss commented on src/components/header/Filter.vue:
```diff
@@ -0,0 +1,32 @@
+    <div
+      class="flex items-center desktop:justify-end phone:sutify-start phone:mt-2 w-full"
+    >
+      <ArrowButton class=""></ArrowButton>
```
Please remove the empty class attribute.

kincsokiss commented on src/components/todo/ToDoList.vue:
```diff
@@ -4,6 +4,9 @@
       <Header @toggle-new-todo="toggleNewTodo"></Header>
 
       <Searchbar class="mb-15" @set-search-query="setSearcQuery" />
```
There is a misspelling issue in `setSearcQuery`. It has to be `setSearchQuery`.

BiroAron committed:
ITL 25 - Implement the ability to sort by title

BiroAron committed:
ITl 26 - Implement the ability to sort by due date

BiroAron committed:
ITL 27 - Implement the ability to sort by priority

BiroAron committed:
ITL 28, ITL 29 - Implement the sort in ascending/descending order

BiroAron committed:
ITL 6 - Made some additional changes

- saving todos list to localstorage
- hiding searchbar and filtering buttons when they aren't necessary

gerebdavid commented on src/components/header/ArrowButton.vue:
```diff
@@ -0,0 +1,24 @@
+<template>
+  <button
+    class="bg-black py-2 px-3 rounded-md font-semibold flex items-center"
+    :class="setBgColor"
```
This should be getBgColor

gerebdavid commented on src/components/header/ArrowButton.vue:
```diff
@@ -0,0 +1,24 @@
+    class="bg-black py-2 px-3 rounded-md font-semibold flex items-center"
+    :class="setBgColor"
+  >
+    <DownArrowIcon v-if="sortAscending"></DownArrowIcon>
```
Use v-if v-else

gerebdavid commented on src/components/header/Filter.vue:
```diff
@@ -0,0 +1,68 @@
+  (e: 'toggleSortOrder'): void
+}>()
+
+const activeortButtonColor = computed(() =>
```
There is a typo

gerebdavid commented on src/components/header/Filter.vue:
```diff
@@ -0,0 +1,68 @@
+  </div>
+</template>
+
+<script setup lang="ts">
```
Please follow the style guide, and move computeds before functions

gerebdavid commented on src/components/todo/ToDoList.vue:
```diff
@@ -76,11 +69,78 @@ const filteredTodos = computed(() => {
+
+  if (sortAscending.value) {
+    return dateA.getTime() - dateB.getTime()
+  } else {
```
dont need else if you return inside if

gerebdavid commented on src/components/todo/ToDoList.vue:
```diff
@@ -76,11 +69,78 @@ const filteredTodos = computed(() => {
+
+  if (sortAscending.value) {
+    return priorityA - priorityB
+  } else {
```
same

BiroAron committed:
ITL 6 - Made some changes regarding the comments

- corrected typos in parameter names
- deleted unnecessary else statements
- restructured Filter.vue according to the style guide
- used v-if v-else where possible

BiroAron committed:
ITL 6 - Made some additional changes

 - corrected priority buttons
- modified Todo type: todo.text is now todo.description

BiroAron committed:
ITL 6 - Added hover effect to todo elements, fixed delete blur bug

BiroAron committed:
ITL 6 -  Corrected css

BiroAron committed:
ITL 6 - Corrected some more css and fixed a v-if
